### DISCUSSION TITLE
What is the toolchain for this project?

### DISCUSSION THREAD

**Discussion by ZolOnTheNet** [Mezryss/FVTT-Genesys#140] [Q&A] [answered]
I'm starting to know javascript but not typescript.
How can we combine the program to obtain a single file? What software suites are required to compile and test the application properly? Are you using vscode or Miscrosoft visual studio? 
I can get some .js but not much more... and for view, I've got nothing...
Do you have any tutorials or links to help me do the job properly?
thank you in advance

best regards

**Comment by Assembling-Kings** [comment, marked as answer]
1. Install Visual Studio Code (any other IDE also works if so desired)
2. Clone the repository locally
3. Make sure you have `node` installed
4. Make sure you have `yarn` installed
5. Open the project on VSCode and on the terminal run `yarn install`
6. To run the test server make sure you have FoundryVTT running, then run `npm run dev`.
7. Now go to your browser and head to http://localhost:30001/

From then on any changes you do on the code will be hot-reloaded into the world. Most of the time it will work just fine but editing some files, like language files, might require you to force a refresh or go out and in of the world.

**Reply by Jackman97** [reply]
Hello and thanks for the explanation.
I've tried using Visual Studio Code but I get the error ECONNREFUSED following the steps. I've searched and haven't found a possible reason or solution.
![image](https://github.com/user-attachments/assets/3ef3ae12-631e-41c9-b184-79028ea86b9d)

**Reply by Assembling-Kings** [reply]
I'm not sure what that error means but from the paths I see on that error message let me clarify two things:
- Clone the repository in a folder outside of your FoundryVTT installation folder. This repository doesn't have the appropriate structure to mimic a system install.
- Install a version of the system in your FoundryVTT installation the normal way.

Let me know if that helps.

**Reply by Jackman97** [reply]
Thanks for the reply. I have tried putting it somewhere else, with the system installed in Foundry (and Foundry open with a Genesys world) and entering the port that Visual Studio Code tells me. But it still doesn't work.
I have tried other programs with Vite and they work fine. Maybe I don't know how to use it.
Sorry for the inconvenience.
![image](https://github.com/user-attachments/assets/186aa344-9f65-4c38-b9a0-0f1aa1822d6f)

**Reply by Assembling-Kings** [reply]
Oh, actually check if anything else is connected to that port and kill it. I should have payed more attention to what you sent initially 😓

**Reply by Jackman97** [reply]
I've tried it but the only port used is the node.js one

**Reply by Assembling-Kings** [reply]
Nothing comes to mind with regards to what might be the problem since that's the process in a nutshell. Maybe try to describe step by step what are you doing without leaving any detail out.

**Reply by Jackman97** [reply]
Thank
1. I run Foundry VTT
2. I clone the repository with the git clone command in my user folder

![image](https://github.com/user-attachments/assets/f5e3e85a-a718-4fcf-9b38-615b074dc36a)

3. I open Visual Studio Code and the cloned folder.

![image](https://github.com/user-attachments/assets/2a536f2b-3734-48d3-9eb2-da00ddd9a2be)

4. I open a new terminal and check that I have node and yarn installed. And I install yarn
![image](https://github.com/user-attachments/assets/17a948ad-6fab-4056-982c-665cca25ea6a)
![image](https://github.com/user-attachments/assets/c998f77b-2c82-459e-85b2-69556d50725c)

5. I run npm run dev and enter the port that appears in the terminal, http://localhost:30001/systems/genesys. I get this page with the 500 error
![image](https://github.com/user-attachments/assets/122bf832-3526-4486-85b6-c3993863ea69)
![image](https://github.com/user-attachments/assets/de4014b2-b6f3-4679-9a27-994cbf7ddece)
![image](https://github.com/user-attachments/assets/472beaa4-5663-4b6a-a815-7b664c0c2651)

6. My Foundry port is another one, I don't know if it has something to do with it.
![image](https://github.com/user-attachments/assets/7ea5dfa1-940f-41d5-83a3-be49bb8d37ee)

**Reply by Assembling-Kings** [reply]
Yeah, the port you use for Foundry does matter. Change this to the port you're using: https://github.com/Mezryss/FVTT-Genesys/blob/b8a40225d1b09776f05db2be5adba5550cd40ac3/vite.config.ts#L12

Or create an environment file with the port you're using as the variable being referenced there.

**Reply by Jackman97** [reply]
That was it, thank you very much and sorry for the inconvenience.
